src/main/java/com/google/api/codegen/transformer/SchemaTypeFormatter.java

```diff
@@ -26,12 +26,16 @@
   /** Get the full name for the given short name, using the default package. */
   String getImplicitPackageFullNameFor(String shortName);
 
-  /** Get the full name for the given type. */
-  String getFullNameFor(Schema type);
+  /** Returns the full name for the given type (without adding the full name to the import set). */
+  String getFullNameFor(Schema type, String parentName);
```

Comment:
Is it possible to get the value of `parentName` from the schema instead of piping it through?